core/templates/pages/exploration-editor-page/services/autosave-info-modals.service.ts

```diff
@@ -17,88 +17,86 @@
  * on the type of response received as a result of the autosaving request.
  */
 
-require('domain/utilities/url-interpolation.service.ts');
-require('pages/exploration-editor-page/services/exploration-data.service.ts');
-require('services/local-storage.service.ts');
-require(
-  'pages/exploration-editor-page/modal-templates/' +
-  'save-validation-fail-modal.controller.ts');
-require(
-  'pages/exploration-editor-page/modal-templates/' +
-  'save-version-mismatch-modal.controller.ts');
-require(
-  'pages/exploration-editor-page/modal-templates/' +
-  'lost-changes-modal.controller.ts');
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
 
-angular.module('oppia').factory('AutosaveInfoModalsService', [
-  '$uibModal', 'LocalStorageService', 'UrlInterpolationService',
-  function(
-      $uibModal, LocalStorageService, UrlInterpolationService) {
-    var _isModalOpen = false;
+import { ExplorationDataService } from 'pages/exploration-editor-page/services/exploration-data.service';
+import { LocalStorageService } from 'services/local-storage.service';
+import { SaveVersionMismatchModalComponent } from '../modal-templates/save-version-mismatch-modal.component';
+import { SaveValidationFailModalComponent } from '../modal-templates/save-validation-fail-modal.component';
+import { LostChangesModalComponent } from '../modal-templates/lost-changes-modal.component';
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
+import { LostChange } from 'domain/exploration/LostChangeObjectFactory';
 
-    return {
-      showNonStrictValidationFailModal: function() {
-        $uibModal.open({
-          templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
-            '/pages/exploration-editor-page/modal-templates/' +
-            'save-validation-fail-modal.template.html'),
-          // Prevent modal from closing when the user clicks outside it.
-          backdrop: 'static',
-          controller: 'SaveValidationFailModalController'
-        }).result.then(function() {
-          _isModalOpen = false;
-        }, function() {
-          _isModalOpen = false;
-        });
+@Injectable({
+  providedIn: 'root'
+})
+export class AutosaveInfoModalsService {
+  _isModalOpen: boolean = false;
```

Comment:
Instead of _ lets actually say private if we want it to be private

Reply:
done!